docs/split-inputs.md

```diff
@@ -119,9 +119,70 @@ It is relatively easy for a modern fuzzing engine to discover the separator,
 but nevertheless we recommend to provide several seed inputs with the desired number
 of separators.
 
-## Fuzz Data Provider
-
-TODO
+## Fuzzed Data Provider
+
+[FuzzedDataProvider] is a single-header C++ library that is helpful for
+splitting a fuzz input into multiple parts of various types. Both advantage and
+disadvantage of using this library is that the input splitting happens
+dynamically, i.e. you don't need to define any structure of the input. This
+might be very helpful in certain cases, but could also make the corpus human
+unreadable.
+
+[FuzzedDataProvider] is a class whose constructor accepts `const uint8_t,
+size_t` arguments. Below is a quick overview of the available methods.
+
+### Methods for extracting individual values
+
+* `ConsumeBool`, `ConsumeIntegral`, `ConsumeIntegralInRange` methods are helpful
+  for extracting a single boolean or integer value, e.g. some flag for the
+  target API, or a number of iterations for a loop, or length of a part of the
+  fuzz input.
+* `ConsumeEnum` and `PickValueInArray` methods are typically applied for the
+  same purpose as the methods above, but are handy when the fuzz input needs to
+  be selected from a predefined set of values, such as an enum or an array.
+
+These methods are using the last bytes of the fuzz input for deriving the
```

Comment:
Are these last bytes used to derive other values too? For example, if someone calls `ConsumeEnum` followed by `ConsumeRemainingBytes`, would the last bytes of the input be shared between the two values? If yes, are there any concerns re. the correlation introduced between the two values?

Reply:
Nope, these bytes are discarded and nothing is used more than once. That's why I had to say that valid inputs can be used as a seed corpus in "the vast majority of use cases" and not "always". Do you think it'd better to add another sentence about this?

Reply:
Actually, changed "the vast majority of use cases" to "some cases".